### DISCUSSION TITLE
[@types/oracledb] BindDefinition does not accept constants of type DBType

### DISCUSSION THREAD

**Discussion by dhrumit9** [DefinitelyTyped/DefinitelyTyped#68883] [Issues with a @types package] [answered]
## Description

In version `6.2.3` of `@types/oracledb`, the constants were retyped as `DBType` instead of `number`. With that change, the `BindParameter` interface was also updated to accept the new type in it's `type` parameter.

But, the `BindDefinition` interface is still not updated to accept the new types and still expects a number constant. This breaks the application in build step and throws type error shown below.

```txt
Type '{ type: OracleDB.DbType & { num: 2010; name: "DB_TYPE_NUMBER"; columnTypeName: "NUMBER"; oraTypeNum: 2; bufferSizeFactor: 22; }; }' is not assignable to type 'BindDefinition'.
        Types of property 'type' are incompatible.
          Type 'DbType & { num: 2010; name: "DB_TYPE_NUMBER"; columnTypeName: "NUMBER"; oraTypeNum: 2; bufferSizeFactor: 22; }' is not assignable to type 'number | undefined'.ts(2322)
```

### Existing Types

- BindParameter
  ```typescript
  interface BindParamter {
      /* Rest of the properties */
      type?: DbType | number | string | undefined;
  }
  ```
- BindDefinition
  ```typescript
  interface BindDefinition {
      /* Rest of the properties */
      type?: number | undefined;
  }
  ```

### Expected Type

- BindDefinition
  ```typescript
  interface BindDefinition {
      /* Rest of the properties */
      type?: DBType | number | undefined;
  }
  ```

[a comment by an automated account is not shown]

**Comment by dannyb648** [comment]
Thansk @dhrumit9 - Another incident which was caused by someone releasing these changes without the proper process. I'll look at a patch for 6.2.4 and this fix will be in 6.3.0 aswell

**Comment by dhrumit9** [comment]
@dannyb648 Thanks for the prompt reply. Looking forward to the 6.2.4 patch as this just broke our builds.
[Marking the discussion as resolved]

**Comment by dhrumit9** [comment]
Hi @dannyb648, the issue still persists on v6.3.0 and v6.4.0. The `BindDefinition` interface still does not accept the DBType constant.

**Reply by dannyb648** [reply]
Sorry man, slipped through the cracks - think this issue got in after some PRs were approved wrongly. I'll add it all now and get you to review the PR for a 6.4.x fix this morning.

**Reply by dannyb648** [reply]
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/6515044/7c6da39b-88d9-4f2e-ba8b-284373dd7d21)

Im guessing this is what we mean here - this full interface was missing the DbType value as a type for type?

**Reply by dhrumit9** [reply]
Yes, that was missing

**Reply by dannyb648** [reply]
Cool - ill ask Connor to review and get us moving into prod

**Reply by dhrumit9** [reply]
Found some more places to fix the same issue. Have added the references in the PR.

**Reply by dannyb648** [reply]
Thanks - ill update them in a mo

**Comment by Anonymous4078** [comment]


**Reply by dhrumit9** [reply]
Well, at least he is trying to fix them, unlike you who approved breaking changes and created this mess in the first place.

**Comment by dannyb648** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69402 Proposed fix for this issue on 6.4.x line - will backport if @dhrumit9 is happy

**Reply by dhrumit9** [reply]
Thanks, I will check the PR soon. And backport to 6.3.x would be nice as well because that version is also broken.

**Comment by SudevAtST** [comment]
@dannyb648 Can you please merge the PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69402 to resolve this issue?

**Comment by dhrumit9** [comment, marked as answer]
Closing this as the PR is merged. (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69402)
